app/ui/reports_query_hash.py

```diff
@@ -0,0 +1,134 @@
+import plotly.graph_objects as go
+import streamlit as st
+from plotly.subplots import make_subplots
+
+import connection
+import filters
+
+
+def report(
+    bf: filters.BaseFilter,
+    cost_per_credit,
+):
+    _ = connection.execute(
+        "CALL INTERNAL.REPORT_PAGE_VIEW('Query Report Repeated Queries')"
+    )
+
+    labels = connection.execute_with_cache(
+        "select name from internal.labels where group_name is null"
+    )
+
+    addition_filter = ""
+
+    if len(labels) > 0:
+        with bf.container:
+            c1, c2, c3 = st.columns(3)
+            with c1:
+                include_all = st.multiselect("Include All", options=labels)
+            with c2:
+                include_any = st.multiselect("Include Any", options=labels)
+            with c3:
+                exclude_any = st.multiselect("Exclude Any", options=labels)
+
+        for label in include_all:
+            addition_filter += f""" and "{label}" """
+
+        if len(include_any) > 0:
+            addition_filter += " and ("
+            for label in include_any:
+                addition_filter += f""" "{label}" or """
+
+            addition_filter = addition_filter[:-3] + ") "
+
+        if len(exclude_any) > 0:
+            addition_filter += " and not ("
+            for label in exclude_any:
+                addition_filter += f""" "{label}" or """
+
+            addition_filter = addition_filter[:-3] + ") "
+
+    sql = f"""
+        with agg as (
+select sum(cost) as cost, count(*) as cnt, any_value(query_text) as query, query_parameterized_hash
+from reporting.labeled_query_history where query_parameterized_hash is not null and cost >0
+        and start_time between %(start)s and %(end)s and (array_size(%(warehouse_names)s) = 0
+            OR array_contains(warehouse_name::variant, %(warehouse_names)s))
+        {addition_filter}
+group by query_parameterized_hash
+), buckets as (
+select sum(cost) as cost, sum(cnt) as cnt, width_bucket(log(10, cnt), 0, 7, 7) as bucket from agg group by all
+)
+select cost as "Cost", cnt as "Count", '[' || pow(10,bucket-1) || ', ' || pow(10,bucket) || ')' as "Bucket" from buckets order by "Bucket" desc
+        """
+
+    def overview():
+        df = connection.execute_select(
+            sql + " ;",
+            {"start": bf.start, "end": bf.end, "warehouse_names": bf.warehouse_names},
+        )
+
+        labels = list(df.Bucket.values)
```

Comment:
i wanted to sort the buckets too. They come out of hte query sorted but plotly resorts by itself :-(

Agreed this may not be the best representation. I won't make any changes now but will continue to noodle and we can take feedback at demo day

Reply:
> They come out of hte query sorted but plotly resorts by itself :-(

blarg!

> I won't make any changes now but will continue to noodle and we can take feedback at demo day

Great idea.